idmapping-rest/src/main/java/org/uniprot/api/idmapping/service/impl/IdMappingJobServiceImpl.java

```diff
@@ -74,7 +77,8 @@ public IdMappingJobServiceImpl(
         this.cacheService = cacheService;
         this.pirService = pirService;
         this.jobTaskExecutor = jobTaskExecutor;
-        this.hashGenerator = new HashGenerator();
+        // make it autowired bean TODO
+        this.hashGenerator = new HashGenerator(new IdMappingJobRequestToArrayConverter(), SALT_STR);
```

Comment:
Minor Comment: 
new HashGenerator<>(new IdMappingJobRequestToArrayConverter(), SALT_STR); is more approppriate.
(diamond sign after generic class name)

(Can be seen from IDE suggestions as well)